flag.go

```diff
@@ -825,16 +834,20 @@ func (f *FlagSet) parseSingleShortArg(shorthands string, args []string) (outShor
 		err = f.failf("flag needs an argument: %q in -%s", c, shorthands)
 		return
 	}
-	err = f.setFlag(flag, value, shorthands)
-	return
+
+	origVal := value
+	if defVal {
+		origVal = ""
+	}
+	return outShorts, outArgs, fn(flag, value, shorthands, "-"+shorthands[:1], origVal)
```

Comment:
Make sense. One question, If we move f.setflag() into the callback function. How do you want to deal with the arguments of f.setflag() ?  If don't  move f.setflag(), we can just pass *flag.Flag to the function.